### PR TITLE
[shared_preferences] remove android no-op implementation

### PR DESCRIPTION
**Pull request by cg021**
## Description

Remove the Android directory in the shared_preferences's shared_preferences_web and shared_preferences_macos subdirectories. These are unused implementations and do not need to be included.

## Related Issues

flutter/flutter#46304

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/shared_preferences/shared_preferences_macos/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 ## 0.0.1+8
 
+* Remove Android folder from `shared_preferences_macos`.
+
+## 0.0.1+8
```
Comment (blasten) [inline packages/shared_preferences/shared_preferences_macos/CHANGELOG.md:5]: nit: typo

Comment (cg021) [reply packages/shared_preferences/shared_preferences_macos/CHANGELOG.md:5]: sorry- just fixed.

**Comment by blasten** [comment]
It also needs to update bump the version of shared_preferences_web

**Comment by kmfish** [comment]
FAILURE: Build failed with an exception.
* Where:
Initialization script '/Users/username/sdk/flutter/packages/flutter_tools/gradle/aar_init_script.gradle' line: 15
* What went wrong:
Android property not found.

It will cause build failed when the android directory is removed from the plugin.

**Comment by shenguojun** [comment]
when running command "flutter build aar", it will return "Android property not found." error. In the aar_init_script.gradle file , will check if project has property "android". Please don't delete android folder after correct the aar_init_script.gradle.

**Pull request merged**